crates/kitsune_p2p/types/src/dht_arc.rs

```diff
@@ -247,6 +247,24 @@ impl DhtArc {
         }
     }
 
+    /// Represent an arc as an optional range of inclusive endpoints.
+    /// If none, the arc length is 0
+    pub fn primitive_range(&self) -> Option<(u32, u32)> {
+        let ArcRange { start, end } = self.range();
+        match (start, end) {
+            (Bound::Included(a), Bound::Included(b)) => Some((a, b)),
+            (Bound::Excluded(_), Bound::Excluded(_)) => None,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Same as primitive_range, but "inside-out"
+    pub fn primitive_range_transposed(&self) -> (Option<u32>, Option<u32>) {
```

Comment:
Just a semantic thing, but I don't like the name "transposed" it makes me think you're inverting the arc / selecting all the non-selected arc-space by swapping the start/end values.

Though, I'm having trouble coming up with a better name... maybe the two values are "detached" from each other?

```suggestion
    pub fn primitive_range_detached(&self) -> (Option<u32>, Option<u32>) {
```